pkg/round/size.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package round
+
+import "math"
+
+const (
+	// GB - GigaByte size
+	GB = 1000 * 1000 * 1000
+	// GiB - GibiByte size
+	GiB = 1024 * 1024 * 1024
+
+	// MB - MegaByte size
+	MB = 1000 * 1000
+	// MiB - MebiByte size
+	MiB = 1024 * 1024
+
+	// KB - KiloByte size
+	KB = 1000
+	// KiB - KibiByte size
+	KiB = 1024
+)
+
+// OffBytes converts roundoff the size
+// 1.1Mib will be round off to 2Mib same for GiB
+// size less than 1MiB will be round off to 1MiB.
+func OffBytes(bytes uint64) uint64 {
```

Comment:
Can't we use Kubernetes' `resource` package (e.g. `resource.NewQuantity`)?

Reply:
We can but since we are dealing all the time with "plain" byte count's it's more handy to not convert to a `ressource`. Furthermore you need first to check for the `Scale` in order to pass it to the `RoundUp` function e.g. `RoundUp(resource.Giga)`

Reply:
Better question: Where is this used / required? Maybe the caller could make better use of the `resource` package.

Reply:
This is used to provision rbd images without fragmentation.